pydantic/_internal/_generate_schema.py

```diff
@@ -642,13 +642,11 @@ def _unpack_refs_defs(self, schema: CoreSchema) -> CoreSchema:
         """Unpack all 'definitions' schemas into `GenerateSchema.defs.definitions`
         and return the inner schema.
         """
-
-        def get_ref(s: CoreSchema) -> str:
-            return s['ref']  # type: ignore
-
         if schema['type'] == 'definitions':
-            self.defs.definitions.update({get_ref(s): s for s in schema['definitions']})
-            schema = schema['schema']
+            definitions = self.defs.definitions
+            for s in schema['definitions']:
+                definitions[s['ref']] = s
```

Comment:
```suggestion
                definitions[s['ref']] = s  # type: ignore
```

Reply:
Similar issue to the one mentioned on the previous PR, though this was already handled above with a type ignore so I feel fine about reusing here